### PR TITLE
[RPA-1609] Global variables scheduler migration

### PR DESCRIPTION
**Pull request by Aeri4a**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-desktop/src/core/websocket/auth/auth.service.ts [hunk 1]
```diff
@@ -36,7 +37,12 @@ export class AuthService {
 
         const token = response.data['id_token'];
         orchestratorAxios.defaults.headers.common.Authorization = `Bearer ${token}`;
+        schedulerAxios.defaults.headers.common.Authorization = `Bearer ${token}`;
         this.storageService.setValue('token', token);
+
+        const authUser = (await orchestratorAxios.get('/api/account')).data;
```
Comment (marcin-depcik) [inline runbotics/runbotics-desktop/src/core/websocket/auth/auth.service.ts]: I think that if it throws an error, it won't be caught in any way, maybe it is better to handle it somehow

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/GlobalVariable/GlobalVariable.thunks.ts [hunk 2]
```diff
@@ -1,40 +1,14 @@
...
 
-export const getGlobalVariables = createAsyncThunk<IGlobalVariable[], void, { state: RootState }>(
-    'globalVariables/getGlobalVariables',
-    async () => {
-        const response = await Axios.get('/api/global-variables');
-        return response.data;
-    },
-);
+export const getGlobalVariables = ApiTenantResource
+    .get<IGlobalVariable[]>('globalVariables/getGlobalVariables', 'global-variables');
```
Comment (marcin-depcik) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/GlobalVariable/GlobalVariable.thunks.ts]: What do you think about assigning a resource name like `global-variables` to some const variable or enum?

File: runbotics/runbotics-scheduler/src/scheduler-database/global-variable/dto/update-global-variable.dto.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { z } from 'zod';
+import { GlobalVariableType } from 'runbotics-common';
+
+export const updateGlobalVariableSchema = z.object({
+    name: z.string().max(255),
+    value: z.string(),
+    type: z.nativeEnum(GlobalVariableType),
+    description: z.string().max(255)
+}).partial();
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/global-variable/dto/update-global-variable.dto.ts:9]: did you use `partial()` for the future?

File: runbotics/runbotics-scheduler/src/scheduler-database/global-variable/global-variable.controller.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+@Controller('/api/scheduler/tenants/:tenantId/global-variables')
+export class GlobalVariableController {
+    private readonly logger = new Logger(GlobalVariableController.name);
+
+    constructor(
+        private readonly globalVariableService: GlobalVariableService,
+    ) {}
+
+    @Get() // TODO: pagination & filtering
+    @FeatureKeys(FeatureKey.GLOBAL_VARIABLE_READ)
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/global-variable/global-variable.controller.ts:30]: have you added feature keys yet?

File: runbotics/runbotics-scheduler/src/scheduler-database/global-variable/global-variable.controller.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        @Param('tenantId') tenantId: Tenant['id'],
+        @User() user: UserEntity
+    ) {
+        this.logger.log('REST request to update global variable with id: ', id);
+        return this.globalVariableService.update(tenantId, user, globalVariableDto, id);
+    }
+
+    @Delete(':id')
+    @HttpCode(HttpStatus.NO_CONTENT)
+    async deleteGlobalVariableBy(
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/global-variable/global-variable.controller.ts:83]: I think you forgot to include `FeatureKeys` guard, didn't you?

**Review by marcin-depcik — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-scheduler/src/scheduler-database/global-variable/dto/update-global-variable.dto.ts (hunk 3 above)
Comment (Aeri4a) [reply runbotics/runbotics-scheduler/src/scheduler-database/global-variable/dto/update-global-variable.dto.ts:9]: in order to make whole schema optional

**Review by marcin-depcik — APPROVED** [review]

**Review by bhszewczyk — APPROVED** [review]

**Pull request merged**